### PR TITLE
Feature/CLI_TX-KEA

### PR DESCRIPTION
**Pull request by AngelicaLastra**
## Description & motivation
This earthmover bundle maps CLI TX-KEA Assessment Results.
Please see earthmover.yml comments for more insights on transformations.

## PR Merge Priority:
- [x] Low
- [ ] Medium
- [ ] High

## Future ToDos & Questions:

**1. Duplicate Columns**
  - Understand the best way to address duplicate columns in the report.csv source. So far, there has not been a clear path on this. One suggestion from Tom was to list the columns at the "sources" section of the earthmover.yml, but given that CLI TX-KEA has over 300 columns, and most of them are crucial for the transformations, it might be a little too busy for a solution:

```
  sources:
    my_source:
      file: my_file.csv
      header_rows:1
      columns: ['col1', 'col2', 'col2b', 'col3']
```
**2. Header row inconsistency**
  - For Dallas 2019 and 2020 report.csvs, the header row is located on index 1, and for the rest of the years, 2021-2024, the header is located on index 3. We do not currently have other district's data for 2019 and 2020, so it is hard to tell if this is a behavior we expect for other district's historical data or if it is an anomaly. Nonetheless, it is hard to figure out a way to conditionally treat these inconsistencies, there is not a way we can tell EM "hey if you are ingesting data from 2019-2020, then set the header row to 1, and for the rest set it to 3..."

**3. Missing columns across years and districts**
  - There are a number of subject assessments that are missing as columns from different districts and years, making the report.csvs inconsistent throughout. This is a major issue since if the columns are not found on the report.csv, it might make EM's transformations fail with the error "column not found in the data set":
```
Dallas 2019: header is 1
    CP_Letter_Sound_spa_v1
    CP_Letter_Sound_v1
    CPM_Admin_Mode
    CPM_Admin_Mode_Date
    CPM_Admin_Mode_Sp   
    CPM_Admin_Mode_Sp_Date
Dallas 2020: header is 1
    CP_Letter_Sound_spa_v1
    CP_Letter_Sound_v1
    CP_Rapid_Vocabulary_spa_v1_RV_set3_spa_v1
    CP_Rapid_Vocabulary_v1_RV_set3_v1
    CPM_Admin_Mode
    CPM_Admin_Mode_Date
    CPM_Admin_Mode_Sp
    CPM_Admin_Mode_Sp_Date
Dallas 2024: 
    CP_Letter_Sound_spa_v1
    CP_Letter_Sound_v1
    CP_Rapid_Vocabulary_spa_v1_RV_set2_spa_v1
    CP_Rapid_Vocabulary_spa_v1_RV_set3_spa_v1
    CP_Rapid_Vocabulary_v1_RV_set2_v1
    CP_Rapid_Vocabulary_v1_RV_set3_v1
Everman 2024: (I feel that this file is just incomplete) - there are over 200 columns missing for this file, it might just be incomplete?
```

### CODE REVIEW

File: assessments/CLI_TX_KEA/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+This is an earthmover bundle created from the following Ed-Fi Data Import Tool mapping:
+* **Title**: Texas Kindergarten Entry Assessment (TX-KEA) Assessment Results - API 3.X
+* **Description**: This template maps CLI TX-KEA Assessment Results, providing a comprehensive evaluation of various learning domains crucial for academic success at kindergarten entry and throughout the kindergarten year.
+    
+    For more info on these data files, see https://public.cliengage.org/tools/assessment/tx-kea/
+
+* **API version**: 5.3
```
Comment (AngelicaLastra) [inline assessments/CLI_TX_KEA/README.md:7]: @rlittle08 how do I decipher the right API Version to use, both on this line of code, and on line [#2](https://github.com/edanalytics/earthmover_edfi_bundles/pull/34/files#diff-94fa5636ac292196ced785a39a40b54de0c6af32d419979615cdad0fca38fa13R2)?

File: assessments/CLI_TX_KEA/earthmover.yml [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+      'colname': 'txkea_language_composite',
+      'descriptor': 'Language Composite',
+      'subject_assess_type':'composite',
+      'subject_assessments': [
+        {'subject_assess_ref_1':[{'subject_asses_ref_colname':'txkea_vocabulary_1', 'subject_asses_ref_descriptor':'Vocabulary Wave 1'}]},
+        {'subject_assess_ref_2':[{'subject_asses_ref_colname':'txkea_listening_comp_1', 'subject_asses_ref_descriptor':'Listening Comprehension 1'}]}
+      ],
+      'subject_assess_lang': 'English',
+      'has_cp_col': False
+    },
```
Comment (AngelicaLastra) [inline assessments/CLI_TX_KEA/earthmover.yml:179]: These subjects ("screener" and "composite") reference scores from other assessments that are part of CLI TX-KEA. Using the scores from these other assessments, it is able to evaluate a student's progress on literacy, language, and potential for dyslexia. In this earthmover.yml, I am treating these "screener" and "composite" subjects as their own unique categories of subjects with colnames "screener_assess_ref_x_score" and "composite_assess_ref_x_score". 

In the Assessments.jsont, these subjects are treated as subject assessments, with an added "type" descriptor designating them as "screener" and "composite" respectively. The CLI TX-KEA assessments they reference are treated as Objective Assessments. I am not sure if this is the best way to approach this, so I am open to suggestions:

```
["studentAssessmentIdentifier": "...kea_dyslexia_screener",
     ...
    "scoreResults": [
      {"assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Score",
        "resultDatatypeTypeDescriptor": "uri://ed-fi.org/ResultDatatypeTypeDescriptor#Integer",
        "result": "2" },
      ...
      { "assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Type",
        "resultDatatypeTypeDescriptor": "uri://ed-fi.org/ResultDatatypeTypeDescriptor#Level",
        "result": "screener"}]
        
    "studentObjectiveAssessments": [
        { "objectiveAssessmentReference": {....},          
          "scoreResults": [
            {"assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Score",
              "resultDatatypeTypeDescriptor":"uri://ed-fi.org/ResultDatatypeTypeDescriptor#Integer",
              "result": "2" },
            {"assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Subject",
              "resultDatatypeTypeDescriptor": "uri://ed-fi.org/ResultDatatypeTypeDescriptor#Level",
              "result": "Letter Sounds"}] },
        {"objectiveAssessmentReference": {...},          
          "scoreResults": [
            {"assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Score",
              "resultDatatypeTypeDescriptor":"uri://ed-fi.org/ResultDatatypeTypeDescriptor#Integer",
              "result": "2"},
            {"assessmentReportingMethodDescriptor": "uri://www.cliengage.org/TX-KEA/AssessmentReportingMethodDescriptor#Subject",
              "resultDatatypeTypeDescriptor": "uri://ed-fi.org/ResultDatatypeTypeDescriptor#Level",
              "result": "Blending"}]}],]
```

File: assessments/CLI_TX_KEA/earthmover.yml [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+        {'wave_1':[{'colname':'txkea_listening_comp_txkea_listening_comp_1', 'descriptor':'Listening Comprehension 1', 'has_cp_col': True}]},
+        {'wave_2':[
+          {'colname':'txkea_listening_comp_txkea_listening_comp_2_receptive', 'descriptor':'Listening Comp 2 Receptive', 'module': 'Receptive', 'has_cp_col': False}, 
+          {'colname':'txkea_listening_comp_txkea_listening_comp_2_expressive', 'descriptor':'Listening Comp 2 Expressive', 'module': 'Expressive', 'has_cp_col': False}
+        ]},
+        {'wave_3':[
+          {'colname':'txkea_listening_comp_txkea_listening_comprehension_3_receptive', 'descriptor':'Listening Comprehension 3 Receptive', 'module': 'Receptive', 'has_cp_col': True},
+          {'colname':'txkea_listening_comp_txkea_listening_comprehension_3_expressive', 'descriptor':'Listening Comprehension 3 Expressive - Dropped', 'module': 'Expressive', 'has_cp_col': False}
+        ]}
+      ],
```
Comment (AngelicaLastra) [inline assessments/CLI_TX_KEA/earthmover.yml:218]: For context CLI Circle TX-KEA's assessment periods occur at BOY, MOY, and EOY. For each of these intervals, there is a subject column ( subject_1 - representing wave 1, subject_2 - representing wave 2...) and within each wave, there can be a set of objective assessments used to evaluate a student..

^ This structure dictated the framework of this dictionary.

**Review by rlittle08 — APPROVED** [review]

**Comment by rlittle08** [comment]
@jalvord1 can we merge this one?

**Review by jalvord1 — APPROVED** [review]

**Pull request merged**